gitlab-mixin/alerts/alerts.libsonnet

```diff
@@ -0,0 +1,80 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'GitLabAlerts',
+        rules: [
+          {
+            alert: 'GitLabHighJobRegistrationFailures',
+            expr: |||
+              100 * rate(job_register_attempts_failed_total{}[5m]) / rate(job_register_attempts_total{}[5m]) 
+              > %(alertsWarningRegistrationFailures)s
+            ||| % $._config,
+            'for': '0',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              summary: 'Large percentage of failed attempts to register a job.',
+              description:
+                ('{{ printf "%%.2f" $value }}%% of job registrations have failed on {{$labels.instance}}, ' +
+                 'which is above threshold of %(alertsWarningRegistrationFailures)s%%.') % $._config,
+            },
+          },
+          {
+            alert: 'GitLabHighRunnerAuthFailure',
+            expr: |||
+              100 * rate(gitlab_ci_runner_authentication_failure_total{}[5m]) / 
+              (rate(gitlab_ci_runner_authentication_success_total{}[5m]) + rate(gitlab_ci_runner_authentication_failure_total{}[5m])) 
+              > %(alertsWarningRunnerAuthFailures)s
+            ||| % $._config,
+            'for': '0',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              summary: 'Large percentage of runner authentication failures.',
+              description:
+                ('{{ printf "%%.2f" $value }}%% of GitLab runner authentication attempts are failing on {{$labels.instance}}, ' +
+                 'which is above the threshold of %(alertsWarningRunnerAuthFailures)s%%.') % $._config,
+            },
+          },
+          {
+            alert: 'GitLabHigh5xxResponses',
+            expr: |||
+              sum by (instance) (rate(http_requests_total{status=~"^5.*"}[5m])) > %(alertsCritical5xxResponses)s
```

Comment:
For this alert, and the 4xx responses, is there a reason that an absolute value is being used, rather than a percentage?

This would trigger for errors on 1/10 requests or 1/10000 requests.  Both are "bad" but this gives you no sense of scale.

Reply:
Good catch; Definitely makes more sense to have these as percentages. Changed to be percentage based for 4xx and 5xx response alerts.